src/backends/ncnn/ncnnlib.cc

```diff
@@ -0,0 +1,234 @@
+    NCNNLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::~NCNNLib()
+    {
+      delete _net;
+      _net = nullptr;
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void NCNNLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::init_mllib(const APIData &ad)
+    {
+        _net->load_param(this->_mlmodel._params.c_str());
+        _net->load_model(this->_mlmodel._weights.c_str());
+
+        if (ad.has("nclasses"))
+	        _nclasses = ad.get("nclasses").get<int>();
+
+        if (ad.has("threads"))
+            _threads = ad.get("threads").get<int>();
+        else
+            _threads = hardware_concurrency();
+
+        _blob_pool_allocator.set_size_compare_ratio(0.0f);
+        _workspace_pool_allocator.set_size_compare_ratio(0.5f);
+        ncnn::Option opt;
+        opt.lightmode = true;
+        opt.num_threads = _threads;
+        opt.blob_allocator = &_blob_pool_allocator;
+        opt.workspace_allocator = &_workspace_pool_allocator;
+        ncnn::set_default_option(opt);
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void NCNNLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::clear_mllib(const APIData &ad)
+    {
+        (void)ad;
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int NCNNLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::train(const APIData &ad,
+                                        APIData &out)
+    {
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int NCNNLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::predict(const APIData &ad,
+										APIData &out)
+    {
+        TInputConnectorStrategy inputc(this->_inputc);
+        TOutputConnectorStrategy tout;
+        try {
+            inputc.transform(ad);
+        } catch (...) {
+            throw;
+        }
+
+        ncnn::Extractor ex = _net->create_extractor();
+        ex.set_num_threads(_threads);
+        ex.input("data", inputc._in);
+
+        APIData ad_output = ad.getobj("parameters").getobj("output");
+
+        // Get bbox
+        bool bbox = false;
+        if (ad_output.has("bbox") && ad_output.get("bbox").get<bool>())
+            bbox = true;
+
+        // Extract detection or classification
+        if (bbox == true) {
+            ex.extract("detection_out", inputc._out);
+        } else {
+            ex.extract("prob", inputc._out);
+        }
+
+        std::vector<APIData> vrad;
+        std::vector<double> probs;
+        std::vector<std::string> cats;
+        std::vector<APIData> bboxes;
+        APIData rad;
+	
+        // Get confidence_threshold
+        double confidence_threshold = 0.0;
+        if (ad_output.has("confidence_threshold")) {
+	  try {
```

Comment:
This try/catch can be replace with
apitools::get_float(ad_output, "confidence_threshold", confidence_threshold);